cudamapper/src/cudamapper_utils.cpp

```diff
@@ -8,62 +8,143 @@
-            fused_overlap.target_end_position_in_read_ = next_overlap.target_end_position_in_read_;
+            expected_chars += get_size<int32_t>(cigar[i]);
         }
-        else
+        // if there is not enough space in buffer reallocate
+        if (get_size<int64_t>(buffer) - chars_in_buffer < expected_chars)
         {
-            set_relative_strand(fused_overlap);
-            fused_overlaps.push_back(fused_overlap);
-            fused_overlap = unfused_overlaps[i + 1];
+            buffer.resize(buffer.size() * 2 + expected_chars);
         }
+        // Add basic overlap information.
+        const int32_t added_chars = std::sprintf(buffer.data() + chars_in_buffer,
+                                                 "%s\t%lu\t%i\t%i\t%c\t%s\t%lu\t%i\t%i\t%i\t%ld\t%i",
+                                                 query_read_name.c_str(),
+                                                 query_parser.get_sequence_by_id(overlaps[i].query_read_id_).seq.length(),
+                                                 overlaps[i].query_start_position_in_read_,
+                                                 overlaps[i].query_end_position_in_read_,
+                                                 static_cast<unsigned char>(overlaps[i].relative_strand),
+                                                 target_read_name.c_str(),
+                                                 target_parser.get_sequence_by_id(overlaps[i].target_read_id_).seq.length(),
+                                                 overlaps[i].target_start_position_in_read_,
+                                                 overlaps[i].target_end_position_in_read_,
+                                                 overlaps[i].num_residues_ * kmer_size, // Print out the number of residue matches multiplied by kmer size to get approximate number of matching bases
+                                                 std::max(std::abs(static_cast<std::int64_t>(overlaps[i].target_start_position_in_read_) - static_cast<std::int64_t>(overlaps[i].target_end_position_in_read_)),
+                                                          std::abs(static_cast<std::int64_t>(overlaps[i].query_start_position_in_read_) - static_cast<std::int64_t>(overlaps[i].query_end_position_in_read_))), //Approximate alignment length
+                                                 255);
+        chars_in_buffer += added_chars;
+        // If CIGAR string is generated, output in PAF.
+        if (!cigar.empty())
+        {
+            const int32_t added_cigar_chars = std::sprintf(buffer.data() + chars_in_buffer,
+                                                           "\tcg:Z:%s",
+                                                           cigar[i].c_str());
+            chars_in_buffer += added_cigar_chars;
+        }
+        // Add new line to demarcate new entry.
+        buffer.data()[chars_in_buffer] = '\n';
```

Comment:
Optional: While correct `buffer.data()[chars_in_buffer]` instead of `buffer[chars_in_buffer]` looks funny. Probably not intentional...

Reply:
Yup, probably remained from the time I was playing around with it.
Done